src/engine.rs

```diff
@@ -350,13 +351,34 @@ where
     }
 
     /// Dumps all operations.
-    #[allow(unused_variables)]
-    pub fn dump_with_file_builder(
-        path: &Path,
-        raft_groups: &[u64],
-        file_builder: Arc<B>,
-    ) -> Result<Vec<LogItem>> {
-        todo!()
+    pub fn dump_with_file_builder(path: &Path, file_builder: Arc<B>) -> Result<LogItemReader<B>> {
+        if !path.exists() {
+            return Err(Error::InvalidArgument(format!(
+                "raft-engine directory or file '{}' does not exist.",
+                path.to_str().unwrap()
+            )));
+        }
+
+        if path.is_dir() {
+            LogItemReader::new_directory_reader(file_builder, path)
+        } else {
+            LogItemReader::new_file_reader(file_builder, path)
+        }
+    }
+}
+
+pub struct DumpIterator<B: FileBuilder> {
+    item_reader: LogItemReader<B>,
+}
+
+impl<B: FileBuilder> Iterator for DumpIterator<B> {
+    type Item = LogItem;
+
+    fn next(&mut self) -> Option<Self::Item> {
+        if let Some(Ok(item)) = self.item_reader.next() {
+            return Some(item);
+        }
+        None
```

Comment:
Those are not needed anymore.

Reply:
@tabokie  We do not need this iterator any more ?

Reply:
You already implemented it for `LogItemReader`.